docs/current/sdk/go/snippets/gitlab-google-cloud/main.go

```diff
@@ -0,0 +1,96 @@
+package main
+
+import (
+    "context"
+    "fmt"
+    "os"
+
+    "dagger.io/dagger"
+    run "cloud.google.com/go/run/apiv2"
+    runpb "cloud.google.com/go/run/apiv2/runpb"
+
+)
+
+const GCR_SERVICE_URL = "projects/PROJECT/locations/us-central1/services/myapp"
+const GCR_PUBLISH_ADDRESS = "gcr.io/PROJECT/myapp"
+
+func main() {
+    // create Dagger client
+    ctx := context.Background()
+    daggerClient, err := dagger.Connect(ctx, dagger.WithLogOutput(os.Stdout))
+    if err != nil {
+        panic(err)
+    }
+    defer daggerClient.Close()
+
+    // get working directory on host
+    source := daggerClient.Host().Directory(".", dagger.HostDirectoryOpts{
+      Exclude: []string{"ci"},
+    })
+
+    // build application
+    builder := daggerClient.Container().
+        From("golang:1.19").
+        WithMountedDirectory("/src", source).
+        WithWorkdir("/src").
+        WithEnvVariable("CGO_ENABLED", "0").
+        WithExec([]string{"go", "build", "-o", "myapp"})
+
+    // add binary to alpine base
+    prodImage := daggerClient.Container().
```

Comment:
@kpenfound how do I instruct building specifically for "linux/amd64" here? See https://github.com/dagger/dagger/issues/4524

Reply:
It would be: 
`.Container(dagger.ContainerOpts{Platform: dagger.Platform{"linux/amd64"}})`
it _might_ work like this too:
`.Container(dagger.ContainerOpts{Platform: "linux/amd64"})`, I can't remember off the top of my head

Reply:
> it _might_ work like this too:
> `.Container(dagger.ContainerOpts{Platform: "linux/amd64"})`, I can't remember off the top of my head

this also works :tada: